src/server/serve.rs

```diff
@@ -435,6 +437,50 @@ fn parse_resolv_conf(content: &str) -> AardvarkResult<Vec<SocketAddr>> {
     Ok(nameservers)
 }
 
+async fn monitor_upstream_resolvers(nameservers: Arc<Mutex<Vec<SocketAddr>>>) {
+    let mut inotify = match Inotify::init() {
+        Ok(inotify) => inotify,
+        Err(err) => {
+            error!("Failed to initialize inotify: {err}");
+            warn!("Nameservers will not be updated on resolv.conf change");
+            return;
+        }
+    };
+    if let Err(err) = inotify
+        .watches()
+        .add(
+            "/etc/resolv.conf",
+            WatchMask::MODIFY,
+        ) {
+        error!("Failed to add watch on /etc/resolv.conf: {err}");
+        warn!("Nameservers will not be updated on resolv.conf change");
+        return;
+    }
+        let mut buffer = [0; 1024];
+        loop{
+            let events = match inotify.read_events_blocking(&mut buffer) {
```

Comment:
it seems inotify has already an async API for tokio so it would be better if we can use that to avoid spawning a blocking task. 
https://docs.rs/inotify/latest/inotify/struct.EventStream.html

note you must add the `stream` feature to the Cargo.toml for this dep.

basically create the eventstream in serve() and the in the main loop where we call `signals.recv().await` that should be switched to a `tokio::select! {}` where we can wait for both the signal and inotify events in "parallel"